data-prepper-api/src/test/java/org/opensearch/dataprepper/model/codec/JsonDecoderTest.java

```diff
@@ -60,15 +69,93 @@ void test_basicJsonDecoder_withTimeReceived() {
         try {
             jsonDecoder.parse(new ByteArrayInputStream(inputString.getBytes()), now, (record) -> {
                 receivedRecord = record;
-                receivedTime = ((DefaultEventHandle)(((Event)record.getData()).getEventHandle())).getInternalOriginationTime();
+                receivedTime = record.getData().getEventHandle().getInternalOriginationTime();
             });
         } catch (Exception e){}
-        
+
         assertNotEquals(receivedRecord, null);
         Map<String, Object> map = receivedRecord.getData().toMap();
         assertThat(map.get("key1"), equalTo(stringValue));
         assertThat(map.get("key2"), equalTo(intValue));
         assertThat(receivedTime, equalTo(now));
     }
 
+    @Nested
+    class JsonDecoderWithInputConfig {
+        private ObjectMapper objectMapper;
+        final List<String> include_keys = new ArrayList<>();
```

Comment:
Rename to `includeKeys`.

Also, set this list in the `@BeforeEach`. Otherwise, tests may interfere with each other.